ruby/bob.rb

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env ruby
+
+class Bob
+
+	def hey (prompt)
+		if nothing(prompt)
+			return "Fine. Be that way!"
+		elsif yelling(prompt)
+			return "Whoa, chill out!"
+		elsif question(prompt)
+			return "Sure."
+		else
+			return "Whatever."
+		end
+	end
+
+	def question(prompt)
+		if prompt[-1, 1] == "?"
+			return true
+		end
+		return false
+	end
+
+	def yelling(prompt)
+		if prompt == prompt.upcase && prompt =~ /[a-zA-Z]/
+			return true
+		end
+		return false
+	end
+
+	def nothing(prompt)
+		if prompt.strip.empty?
+			return true
+		end
+		return false
+	end
+
+end
```

Comment:
Overall, this looks really good! The only small change I'd suggest is to make use of Ruby's implicit returns for your test methods (you can get rid of all of your conditionals).